File: web/lib/db/audit-events.ts
import { createClient } from '@/lib/supabase/server'
import type { Database } from '@/lib/db/types'

type AuditEventInsert = Database['public']['Tables']['audit_events']['Insert']

// CRITICAL: This function MUST NEVER throw. Audit logging must never break
// the main application flow. All errors are silently swallowed.
export async function insertAuditEvent(event: AuditEventInsert): Promise<void> {
  try {
    const supabase = await createClient()
    await supabase.from('audit_events').insert(event)
  } catch {
    // Audit must never break the main flow — silent swallow
  }
}

File: worker/db/audit_events.py
from typing import Any

import structlog
from supabase import AsyncClient

log = structlog.get_logger()


async def insert_audit_event(client: AsyncClient, event: dict[str, Any]) -> None:
    """Audit logging must never break the main flow — all errors swallowed."""
    try:
        await client.table("audit_events").insert(event).execute()
    except Exception as exc:  # noqa: BLE001 — audit must never raise
        log.warning("audit_event_insert_failed", error=str(exc))
